### ISSUE TITLE
Feature request, code lens run in release mode

### ISSUE THREAD

**Issue by jacobrosenthal** [rust-analyzer/rust-analyzer#5825]
My runner is generally dfu a binary to a device so Id like to be in release mode. I suppose I could just override the debug release mode, but it feels meh.

I searched config a bit and thought I had to ideas, but they didnt seem to work

rust-analyzer.cargoRunner takes an extension name, not a local exe so I cant sneak cargo run --release there

I realized I could sneak an environvariable with runnableEnv but I think its not being injected to cargo, only the result
"rust-analyzer.runnableEnv": {
    "PROFILE": "release"
}

Thoughts?

**Comment by eminence** [comment]
I'm also interested in this.  

Two possible implementations come to mind:

* A configuration change that will change the "Run" lens to run in release mode.  
* Adding a new lens for a run in release mode, so we would see 3 options above the `fn main()`:

> 🢒 Run | Run Release | Debug

**Comment by yay** [comment]
Yes please!! Release build can make a huge difference for GUI apps, where it's not mostly about IO performance.

**Comment by yay** [comment]
I find that for me this is good enough. With this the `Run` lens runs with `--release` flag and if you want to run with debug just run the `Debug` lens with no breakpoints.
![image](https://github.com/rust-lang/rust-analyzer/assets/1799756/068d553e-6c8c-4131-9fd6-327766edb384)

**Comment by lnicola** [comment]
CC https://github.com/rust-lang/rust-analyzer/issues/16047

@yay as a workaround, you can also tweak the build mode in [Cargo.toml](https://doc.rust-lang.org/cargo/reference/profiles.html#overrides).
